src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java

```diff
@@ -902,104 +920,72 @@ public void logOut(String wikiURL, String articleName) {
+
+      HttpClient httpclient = HttpClientBuilder.create()
+          .setConnectionBackoffStrategy(new DefaultBackoffStrategy())
+          .setDefaultCookieStore(cookieStore)
+          .disableAutomaticRetries()
+          .build();
+      HttpPost httpPost = new HttpPost(heliosBaseUrl);
+      List<NameValuePair> nvps = new ArrayList<NameValuePair>();
+
+      nvps.add(new BasicNameValuePair("grant_type", HeliosConfig.GrantType.PASSWORD.getGrantType()));
+      nvps.add(new BasicNameValuePair("client_id", client_id));
+      nvps.add(new BasicNameValuePair("client_secret", client_secret));
+      nvps.add(new BasicNameValuePair("username", userName));
+      nvps.add(new BasicNameValuePair("password", password));
+
+      httpPost.setEntity(new UrlEncodedFormEntity(nvps, StandardCharsets.UTF_8));
+
+      HttpResponse response = httpclient.execute(httpPost);
+      HttpEntity entity = response.getEntity();
+
+      JSONObject responseValue = new JSONObject(EntityUtils.toString(entity));
+
+      PageObjectLogging.log("LOGIN HEADERS: ", response.toString(), true);
+      PageObjectLogging.log("LOGIN RESPONSE: ", responseValue.toString(), true);
+
+      String token = responseValue.getString("access_token");
+      String refreshToken = responseValue.getString("refresh_token");
+      System.out.println("access_token: " + token);
+      System.out.println("refresh_token: " + refreshToken);
+
+      driver.manage().addCookie(new Cookie("access_token", token, ".wikia.com", null, null));
+
+      try {
+        driver.get(wikiURL);
+        System.out.println("cookie: " + driver.manage().getCookieNamed("access_token"));
+
+      } catch (TimeoutException e) {
+        PageObjectLogging.log("loginCookie",
+                              "page timeout after login by cookie", true);
+      }
+
       verifyUserLoggedIn(userName);
-//
-//      PageObjectLogging.log("loginCookie",
-//                            "user was logged in by cookie", true, driver);
-//      return token;
-//    } catch (UnsupportedEncodingException e) {
-//      PageObjectLogging.log("logInCookie",
-//                            "UnsupportedEncodingException", false);
-//      return null;
-//    } catch (ClientProtocolException e) {
-//      PageObjectLogging.log("logInCookie", "ClientProtocolException",
-//                            false);
-//      return null;
-//    } catch (ParseException e) {
-//      PageObjectLogging.log("logInCookie", e.getMessage(), false);
-//      return null;
-//    } catch (IOException e) {
-//      PageObjectLogging.log("logInCookie", e.getMessage(), false);
-//      return null;
-//    } catch (JSONException e) {
-//      e.printStackTrace();
-//      return null;
-//    }
-    return "";
+      PageObjectLogging.log("loginCookie",
+                            "user was logged in by cookie", true, driver);
+      return token;
+    } catch (UnsupportedEncodingException e) {
+      PageObjectLogging.log("logInCookie",
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "logInCookie" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1192)